### ISSUE TITLE
Don't hardcode /usr/include in meson.build

### ISSUE THREAD

**Issue by rossburton** [ximion/appstream#569]
If AppStream is build with stemmer enabled, it forcibly looks in `/usr/include`:

https://github.com/ximion/appstream/blob/1217913bc0276ad5a0bf02b8ab93744c373fc4f4/meson.build#L188

I may have installed stemmer anywhere, and in cross-compiled environments hardcoding this path is absolutely the wrong thing to do as it is the _host_ includes not the _target_ includes.

**Comment by ximion** [comment]
Patches welcome ;-)
Ideally Snowball would just ship a pkg-config file, but annoyingly it doesn't.

**Comment by rossburton** [comment]
Filed https://github.com/snowballstem/snowball/issues/190 to nudge snowball maintainers.

**Comment by eli-schwartz** [comment]
> Ideally Snowball would just ship a pkg-config file, but annoyingly it doesn't.

Ideally snowball would support installing anything at all, period.

However I am slightly curious why you need to specify any include directories at all. Do the compiler builtin defaults not work? That should already have /usr/include, but only on native builds, since in cross builds it will have /path/to/cross-sysroot/usr/include

And given that you don't know where snowball is installed without a pkg-config file, it seems 100% pointless to hardcode an arbitrary location that may not be true.

**Comment by rossburton** [comment]
Right, I'd have thought that the meson header/library checking code would have been sufficient to validate that the library is present.

**Comment by apteryks** [comment]
FWIW, we also hit that one in Guix.
